application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationServiceImpl.java

```diff
@@ -257,10 +259,28 @@ public TraineeRecommendationRecordDto getLatestRecommendation(String gmcId) {
     log.info("Fetching latest recommendation info for GmcId: {}", gmcId);
     Optional<Recommendation> optionalRecommendation = recommendationRepository
         .findFirstByGmcNumberOrderByActualSubmissionDateDesc(gmcId);
+    final DoctorsForDB doctorForDB;
+    final var optionalDoctorsForDB = doctorsForDBRepository.findById(gmcId);
 
-    if (optionalRecommendation.isPresent()) {
-      final var recommendation = optionalRecommendation.get();
+    if (optionalDoctorsForDB.isPresent()) {
+      doctorForDB = optionalDoctorsForDB.get();
+    } else {
+      throw new RecommendationException(format("Doctor %s does not exist!", gmcId));
+    }
 
+    if (optionalRecommendation.isPresent()) {
+      final Recommendation recommendation = optionalRecommendation.get();
+      final RecommendationGmcOutcome outcome = recommendation.getOutcome();
+      final boolean completed = outcome != null
+          && (APPROVED.getOutcome().equals(outcome.getOutcome())
+          || REJECTED.getOutcome().equals(outcome.getOutcome()));
+      //if doctor is under notice but has past completed revalidation -> start new recommendation
+      if (doctorForDB.getUnderNotice().equals(YES)
+          && completed
+          && recommendation.getGmcSubmissionDate() != null
+          && LocalDate.now().isAfter(recommendation.getGmcSubmissionDate())){
```

Comment:
Could move this to a `isNewRecommentionNeeded` (or similar) private method to help keep it a bit tidier/clearer